testutils/generators.go

```diff
@@ -32,16 +36,38 @@ func GenerateRecords(batchSize int) func() (record *exporter.Record, e error) {
 		return &exporter.Record{
 			RecordNumber: uint32(cnt),
 			Record: insrecord.Material{
-				ID:    gen.IDWithPulse(pn),
-				JetID: GenerateUniqueJetID(),
+				Polymorph: int32(randNum()),
+				Virtual: insrecord.Virtual{
+					Polymorph: int32(randNum()),
+					Union:     nil,
+					Signature: []byte{0, 1, 2},
+				},
+				ID:        gen.IDWithPulse(pn),
+				ObjectID:  gen.ID(),
+				JetID:     gen.JetID(),
+				Signature: []byte{0, 1, 2},
 			},
-			ShouldIterateFrom: nil,
+			ShouldIterateFrom: &pn,
+			Polymorph:         uint32(randNum()),
 		}, nil
 	}
 
 	return generateRecords
 }
 
+func GenerateRecordsList(count int) *[]exporter.Record {
+	var res []exporter.Record
+	f := GenerateRecords(count)
+	for {
+		record, err := f()
+		if err != nil {
+			break
```

Comment:
you did not add records to result list, so if no error you will just exit from function and return empty list